wsmux/mux.go

```diff
@@ -0,0 +1,24 @@
+package wsmux
+
+import (
+	"github.com/gorilla/websocket"
+	"time"
+)
+
+type Config struct {
+	RemoteCloseCallback func()
+	AcceptDeadline      time.Time
+	ReadDeadline        time.Time
+	WriteDeadline       time.Time
+	Log                 Logger
+}
+
+// Server (NOTE: set server field to 0 for server connection)
+func Server(conn *websocket.Conn, conf *Config) *Session {
```

Comment:
Don't pass config by pointer... Small structs are cheap to copy... And copying reduces risks of race conditions..